src/slack-model.jl

```diff
@@ -465,16 +466,30 @@ function NLPModels.jac_op_residual!(
 )
   @lencheck nls.meta.nvar x Jtv
   @lencheck nls.nls_meta.nequ Jv
-  prod = @closure v -> jprod_residual!(nls, x, v, Jv)
-  ctprod = @closure v -> jtprod_residual!(nls, x, v, Jtv)
+  prod! = @closure (res, v, α, β) -> begin
```

Comment:
Can you add tests like [these](https://github.com/JuliaSmoothOptimizers/NLPModels.jl/blob/master/test/nlp/api.jl#L89-L91) and [these](https://github.com/JuliaSmoothOptimizers/NLPModels.jl/blob/master/test/nlp/api.jl#L54-L57)